src/components/walletconnect-list/WalletConnectV2ListItem.tsx

```diff
@@ -58,64 +60,49 @@ export function WalletConnectV2ListItem({ session, reload }: { session: SessionT
     },
   };
 
-  const { dappName, dappUrl, dappLogo, address, chainIds } = React.useMemo(() => {
-    const { namespaces, requiredNamespaces, peer } = session;
-    const { metadata } = peer;
-    const chains = requiredNamespaces?.eip155?.chains || [];
-
-    const eip155Account = namespaces.eip155?.accounts?.[0] || undefined;
-
-    if (!eip155Account) {
-      const e = new RainbowError(`[WalletConnectV2ListItem]: unsupported namespace`);
-      logger.error(e);
-
-      // defensive, just for types, should never happen
-      throw e;
-    }
+  const { namespaces, peer } = session;
+  const { metadata } = peer;
 
+  useEffect(() => {
+    const eip155Account = session.namespaces.eip155?.accounts?.[0] || undefined;
     const address = eip155Account?.split(':')?.[2];
-    const chainIds = (chains
-      ?.map(chain => parseInt(chain.split(':')[1]))
-      ?.filter(chainId => supportedWalletConnectChainIds.includes(chainId)) ?? []) as ChainId[];
-
-    if (!address) {
-      const e = new RainbowError(`[WalletConnectV2ListItem]: could not parse address`);
-      logger.error(e);
+    setAddress(address);
+  }, [session]);
 
-      // defensive, just for types, should never happen
-      throw e;
+  useEffect(() => {
+    if (address) {
+      setAccountInfo(getAccountProfileInfo(findWalletWithAccount(wallets || {}, address), walletNames, address));
     }
+  }, [address, walletNames, wallets]);
+
+  const chains = useMemo(() => namespaces?.eip155?.chains || [], [namespaces]);
```

Comment:
i dont think you need useMemo here

Reply:
IDE was telling me I did. I'll leave it for now